src/main.rs

```diff
@@ -81,8 +81,8 @@ fn main_impl(args: ValidatedArgs) -> Result<()> {
     // while also maintaining `tick()` calls to keyberon.
 
     let (server, ntx, nrx) = if let Some(port) = args.port {
```

Comment:
@jtroo The TCPServer (and the notification loop down below) is only started here if the user provides the optional `--port` argument to the CLI 👌